query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -776,6 +778,73 @@ func (PushDownWindowAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (p
 	}), true, nil
 }
 
+//
+// Push Down of window aggregates.
+// ReadRangePhys |> window |> { min, max, mean, count, sum }
+//
+// PushDownWindowAggregateWithTimeRule will match the given pattern.
+// ReadWindowAggregatePhys |> duplicate |> window(every: inf)
+//
+// If this pattern matches and the arguments to duplicate are
+// matching time column names, it will set the time column on
+// the spec.
+type PushDownWindowAggregateByTimeRule struct{}
+
+func (PushDownWindowAggregateByTimeRule) Name() string {
+	return "PushDownWindowAggregateByTimeRule"
+}
+
+func (rule PushDownWindowAggregateByTimeRule) Pattern() plan.Pattern {
+	return plan.Pat(universe.WindowKind,
+		plan.Pat(universe.SchemaMutationKind,
+			plan.Pat(ReadWindowAggregatePhysKind)))
+}
+
+func (PushDownWindowAggregateByTimeRule) Rewrite(ctx context.Context, pn plan.Node) (plan.Node, bool, error) {
+	windowNode := pn
+	windowSpec := windowNode.ProcedureSpec().(*universe.WindowProcedureSpec)
+
+	duplicateNode := windowNode.Predecessors()[0]
+	duplicateSpec, duplicateSpecOk := func() (*universe.DuplicateOpSpec, bool) {
+		s := duplicateNode.ProcedureSpec().(*universe.SchemaMutationProcedureSpec)
+		if len(s.Mutations) != 1 {
+			return nil, false
+		}
+		mutator, ok := s.Mutations[0].(*universe.DuplicateOpSpec)
+		return mutator, ok
+	}()
+	if !duplicateSpecOk {
+		return pn, false, nil
+	}
+
+	// The As field must be the default time value
+	// and the column must be start or stop.
+	if duplicateSpec.As != execute.DefaultTimeColLabel ||
+		(duplicateSpec.Column != execute.DefaultStartColLabel && duplicateSpec.Column != execute.DefaultStopColLabel) {
+		return pn, false, nil
+	}
+
+	// window(every: inf)
+	if windowSpec.Window.Every != values.ConvertDuration(math.MaxInt64) &&
+		windowSpec.TimeColumn == execute.DefaultTimeColLabel &&
+		windowSpec.StartColumn == execute.DefaultStartColLabel &&
+		windowSpec.StopColumn == execute.DefaultStopColLabel &&
+		!windowSpec.CreateEmpty {
+		return pn, false, nil
+	}
+
+	// Cannot rewrite if already was rewritten.
+	windowAggregateNode := duplicateNode.Predecessors()[0]
+	windowAggregateSpec := windowAggregateNode.ProcedureSpec().(*ReadWindowAggregatePhysSpec)
+	if windowAggregateSpec.TimeColumn != "" {
+		return pn, false, nil
+	}
+
+	// Rule passes.
+	windowAggregateSpec.TimeColumn = duplicateSpec.Column
+	return windowAggregateNode, true, nil
```

Comment:
Although not 100% sure, I think this does not remove the `duplicate` and the `window` node, it just replaces the `window` node with a `ReadWindowAggregatePhys` node.

That being said, it transforms
`ReadWindowAggregatePhys |> duplicate |> window(every: inf)`
to 
`ReadWindowAggregatePhys |> duplicate |> ReadWindowAggregatePhys`

I guess you probably need to use `plan.MergeToPhysicalNode()` following the example of `PushDownRangeRule`:
https://github.com/influxdata/influxdb/blob/master/query/stdlib/influxdata/influxdb/rules.go#L195

Reply:
We can verify that after you got the planner test case.